### PR TITLE
Removed `@woocommerce/notices` Internal Prefix

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

In #33218, the `@woocommerce/notices` package was renamed according to the `"private": true` status of the package. Since we're exposing this through the dependency extraction plugin, however, it is actually a public package in some form. Renaming this package would be a backward compatibility break, and as such, we should keep the original name. This pull request changes it back from the `internal-` prefixed version.

### How to test the changes in this Pull Request:

1. Make sure CI passes

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33283/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Removed `@woocommerce/notices` Internal Prefix` a4511d83d28dc04d15367fa7dfd759a34ed44167
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

**Comment by psealock** [comment]
>Since we're exposing this through the dependency extraction plugin, however, it is actually a public package in some form. Renaming this package would be a backward compatibility break, and as such, we should keep the original name.

A way to keep backwards compatibility would be to simply add both `@woocommerce/notices` and `@woocommerce/internal-notices` to the packages Dependency Webpack Extraction Plugin looks out for.

https://github.com/woocommerce/woocommerce/blob/c0619dc597d3e7bf9d97ae46da4f6e09a307248e/packages/js/dependency-extraction-webpack-plugin/assets/packages.js#L15

**Comment by ObliviousHarmony** [comment]
I don't really know much about the dependency extraction plugin @psealock, could you expand on what you mean?

**Comment by psealock** [comment]
Depedency Extraction Webpack Plugin (DEWP) extends [Gutenberg's version](https://github.com/WordPress/gutenberg/blob/trunk/packages/dependency-extraction-webpack-plugin/README.md) and is a tool for handling dependencies in the WP environment.

For example, `@woocommerce/components` is a script available at `window.wc.components`. However, in our code we import like so:

```js
import { Button } from '@woocommerce/components';
```

In order to avoid actually bundling the components, as well as every other plugin also using DEWP, DEWP tells webpack to avoid bundling those imports and gather those functions from the window instead.

If we add `@woocommerce/internal-notices` and `@woocommerce/notices` to the list of packages DEWP knows to create aliases for, then any plugin using `notices` will be able to declare either in imports and DEWP will correctly handle it.

And of course, this begs the question, would an external plugin be using `@woocommerce/notices` in the first place? While it is a `private` package as far as NPM is concerned `window.wc.notices` is available to any and all plugins:

image

The changes in https://github.com/woocommerce/woocommerce/pull/33218 now make the notices package available to all plugins at `window.wc.internal-notices`.

In short, the backwards compatibility play here is to modify DEWP such that either of these work:

```js
import { Notice } from '@woocommerce/notices';
import { Notice } from '@woocommerce/internal-notices';
```
And now that I think about it, this would require an updated DEWP to work so there is a backwards incompatible change introduced. And this leads me to the conclusion that this PR is indeed required.

@ObliviousHarmony, is that conclusion sound?

**Comment by ObliviousHarmony** [comment]
Thanks for the explanation @psealock, this clears everything up. It seems like this PR would be the easiest solution. One possible backwards compatible solution would be to map `@woocommerce/internal-notices` to `wc.notices`, but, I'm not sure what this modification would entail. It would be neat to do that but it may not be worth the effort.

**Comment by psealock** [comment]
>One possible backwards compatible solution would be to map @woocommerce/internal-notices to wc.notices

Thats a good idea. It would involve modifying DEWP, webpack, and `WCAdminAssets.php` to name the module `notices` when the pathname is `internal-notices`.

While I think this is doable, this line leads me to believe the testing required as well as surface area for bug introduction would just be temporary anyways.

https://github.com/woocommerce/woocommerce/blob/29ddd17e6838b022e8010742246b12201f5555d0/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php#L255

So in the end this solution proposed in the PR is best

**Review by psealock — APPROVED** [review]
Changes look good. All tests pass except for changelogger, but since this is reverting an unreleased change, we can ignore.
